src/pages/RepoPage/CoverageTab/subroute/Fileviewer/ComponentsSelectCommit.spec.tsx

```diff
@@ -0,0 +1,186 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query'
+import { render, screen, waitFor } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import { Location } from 'history'
+import { graphql } from 'msw'
+import { setupServer } from 'msw/node'
+import type { ReactNode } from 'react'
+import { MemoryRouter, Route } from 'react-router-dom'
+
+import { useFlags } from 'shared/featureFlags'
+
+import { ComponentsSelectCommit } from './ComponentsSelectCommit'
+
+import { useSummary } from '../../summaryHooks'
+
+jest.mock('shared/featureFlags')
+jest.mock('../../summaryHooks')
+
+const mockedUseFlags = useFlags as jest.Mock<{ componentsSelect: boolean }>
+const mockedUseSummary = useSummary as jest.Mock<any>
```

Comment:
Wondering about the justification for mocking out this hook, over using msw? We typically want to avoid this so we have a deeper integration test, as well if any changes are ever made to `useSummary` that break the functionality we won't pick it up.

Reply:
My initial thoughts were mocking all the hooks within `useSummary()` would have been overkill, but I agree with your point about not picking up changes to `useSummary`. Updated.